tempto-core/src/main/java/com/teradata/tempto/assertions/QueryAssert.java

```diff
@@ -415,6 +415,25 @@ public QueryExecutionAssert failsWithMessage(String expectedErrorMessage)
 
             return this;
         }
+
+        public QueryExecutionAssert failsWithMessageMathing(String expectedErrorMessagePattern)
+        {
+            requireNonNull(expectedErrorMessagePattern, "expectedErrorMessagePattern is null");
+
+            QueryExecutionException executionException = executionExceptionOptional
+                    .orElseThrow(() -> new AssertionError("Query did not fail as expected."));
+
+            String exceptionMessage = executionException.getMessage();
+            LOGGER.debug("Query failed as expected, with message: {}", exceptionMessage);
+            if (!nullToEmpty(exceptionMessage).matches(expectedErrorMessagePattern)) {
+                throw new AssertionError(format(
+                        "Query failed with unexpected error message: '%s' \n Expected error message to match '%s'",
+                        exceptionMessage,
+                        expectedErrorMessagePattern));
+            }
+
+            return this;
```

Comment:
this method is copy-paste of method above, maybe you could extract common code

Reply:
good idea, done